pkg/controller/defaults/placements.go

```diff
@@ -90,6 +90,90 @@ var (
 			},
 		},
 
+		"osd-prepare": rook.Placement{
+			Tolerations: []corev1.Toleration{
```

Comment:
Since the "all" placement has the toleration, you shouldn't need to duplicate it for the OSDs. Although perhaps the storageClassDeviceSets pick up the "all" placement, so you might need it. If you do need it, you could factor out to a helper method so you don't have the same toleration code duplicated.

Reply:
Noted!. as discussed, will plan to refactor it in a separate PR.